test/e2e/upgradetest/recovery_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2017 The etcd-operator Authors
```

Comment:
> It will become a long file by the end with all 5 upgrade test cases.

It doesn't matter :)

> The issue is that the per test setup/cleanup is a little long and is mostly repeated for each case. Perhaps if we just abstract it we can shorten the test cases and put them in one file.

That's orthogonal. It doesn't matter if we put them in one file.

> Let me know if you're still sure we should put all the cases in one file?

yes.